### PR TITLE
docs(mf-parser): add customizable fields into PartInfo

### PR DESCRIPTION
**Pull request by tpoisseau**
I still need to test typings works as expected

### CODE REVIEW

File: packages/mf-parser/src/__tests__/MF.test.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { describe, expect, it } from 'vitest';
 
-import { MF } from '..';
+import { MF } from '../../lib/src';
```
Comment (targos) [inline packages/mf-parser/src/__tests__/MF.test.js:3]: We should never have to import from the built lib folder.

**Comment by targos** [comment]
> I still need to test typings works as expected

There are tools in Vitest to assert types: https://vitest.dev/api/expect-typeof.html#expecttypeof

**Comment by targos** [comment]
> the ones from JSDoc is not properly parsed.

Can we not fix that?

**Comment by tpoisseau** [comment]
This is advanced jsdoc on `@template` (equivalent to typescript `Generic`). It's well supported by typescript, but not well supported by IDE or other tools.

CleanShot 2024-12-05 at 11 12 29@2x

Produce valid .d.ts

```ts
    getInfo<GIO extends import("./util/getInfo.types").GetInfoOptions<string, string> = import("./util/getInfo.types").GetInfoOptions<"monoisotopicMass", "observedMonoisotopicMass">>(options?: GIO): import("./util/getInfo.types").PartInfo<GIO> | import("./util/getInfo.types").PartInfoWithParts<GIO>;
```

An other possibility could be to provide hand-written `.d.ts` aside js files. Or migrate from js to ts.

**Comment by tpoisseau** [comment]
It's more hard than I thought 😬. But it's done for getInfo

File: packages/mf-parser/src/index.js [hunk 2]
```diff
@@ -3,9 +3,10 @@ export * from './util/subSuperscript.js';
 export * from './util/toDisplay.js';
 export * from './util/isMF.js';
 export * from './util/toHtml.js';
-export * from './Kind.js';
+export * from './Kind.ts';
```
Comment (targos) [inline packages/mf-parser/src/index.js:6]: You should keep `.js` even if the file is typescript

**Comment by tpoisseau** [comment]
Too much changes, dead-end, restart here: https://github.com/cheminfo/mass-tools/pull/241

**Pull request closed without merging**